pool/pool.py

```diff
@@ -668,14 +688,10 @@ async def update_farmer(self, request: PutFarmerRequest, metadata: RequestMetada
                 farmer_dict["authentication_public_key"] = request.payload.authentication_public_key
 
         if request.payload.payout_instructions is not None:
-            is_new_value = (
-                farmer_record.payout_instructions != request.payload.payout_instructions
-                and request.payload.payout_instructions is not None
-                and len(hexstr_to_bytes(request.payload.payout_instructions)) == 32
-            )
-            response_dict["payout_instructions"] = is_new_value
-            if is_new_value:
-                farmer_dict["payout_instructions"] = request.payload.payout_instructions
+            new_ph = self.validate_payout_instructions(request.payload.payout_instructions)
```

Comment:
type annotation

Reply:
Actually, it's more readable and less error prone to create 2 new variables here, it's a cool python feature:
`encoded_correctly, puzzle_hash = self.validate.....`

Reply:
Forget what I said above, we can even use just one variable:
`new_ph: Optional[bytes32] = self.validate.....`

And then in the if statement do:
`if new_ph is not None`

Reply:
Also I believe we need to convert back to hexstr as pointed out by dustinface [here,](https://github.com/Chia-Network/pool-reference/pull/176/files) since the farmerDict is json and then gets converted into a streamable.